### ISSUE TITLE
Spectrogram example bug on Ubuntu

### ISSUE THREAD

**Issue by NumesSanguis** [bokeh/bokeh#8779]
Issue related to: https://github.com/bokeh/bokeh/tree/master/examples/app/spectrogram

#### ALL software version info (bokeh, python, notebook, OS, browser, any other relevant packages)
- Bokeh: '1.0.4'   (`conda install bokeh`)
- Python: 3.6.8 (h0371630_0)
- tornado: 5.1.1 (py36h14c3975_1000) (`conda install -c conda-forge tornado`)
- nodejs: 10.13.0  (he6710b0_0) (`conda install nodejs`)
- pyaudio: 0.2.11 (py36h14c3975_1)  (`conda install pyaudio`)
- Ubuntu: 18.04.2 LTS 64-bit
- GNOME: 3.28.2
- Firefox Quantum 66.0 (64-bit)

#### Description of expected behavior and the observed behavior

gio open (Synthesized audio data, no real-time audio data with Pyaudio)

- Expected: `bokeh serve --show spectrogram/` opens a browser tab and shows a spectrogram
- Observed:  `bokeh serve --show spectrogram/` opens a browser tab, but only some text and the 2 sliders "Gain" and "Frequency are shown". Also the error message in the terminal `This tool has been deprecated, use 'gio open' instead.`.

#### Complete, minimal, self-contained example code that reproduces the issue

```
# 1. Commenting out the try: block for pyaudio in audio.py (to test with synthesized data)
# 2. execute the following command in bokeh/examples/app/
bokeh serve --show spectrogram/
```

#### Stack traceback and/or browser JavaScript console output

Terminal output:

```
bokeh serve --show spectrogram/

 *** Pyaudio package not installed, using synthesized audio data ***

2019-03-22 14:24:23,311 Starting Bokeh server version 1.0.4 (running on Tornado 5.1.1)
2019-03-22 14:24:23,313 Bokeh app running at: http://localhost:5006/spectrogram
2019-03-22 14:24:23,313 Starting Bokeh server with process id: 29990
This tool has been deprecated, use 'gio open' instead.
See 'gio help open' for more info.

2019-03-22 14:24:24,522 200 GET /spectrogram (127.0.0.1) 935.95ms
2019-03-22 14:24:24,571 200 GET /spectrogram (127.0.0.1) 47.54ms
2019-03-22 14:24:24,735 101 GET /spectrogram/ws?bokeh-protocol-version=1.0&bokeh-session-id=pHNZ3ATAuQpws2iwMVCtNZDWtIAgvadyw7IFUg1VUsHN (127.0.0.1) 0.44ms
2019-03-22 14:24:24,735 WebSocket connection opened
2019-03-22 14:24:24,735 ServerConnection created
2019-03-22 14:24:27,541 101 GET /spectrogram/ws?bokeh-protocol-version=1.0&bokeh-session-id=V83caWBwhGc7R2JkJEOXkdzqQwuFuPFRFq4mcBp7hob5 (127.0.0.1) 0.67ms
2019-03-22 14:24:27,541 WebSocket connection opened
2019-03-22 14:24:27,541 ServerConnection created
```

Javascript output:

```
[bokeh] setting log level to: 'info'            logging.js:130
[bokeh] Websocket connection 0 is now            open connection.js:183
[bokeh] LogTickFormatter not configured with a ticker, using default base of 10 (labels will be incorrect if ticker base is not 10)            log_tick_formatter.js:22
[bokeh] WidgetBox mode is fixed, but no width specified. Using default of 300.            widget_box.js:98
Error rendering Bokeh items: TypeError: "t is undefined"
    initialize renderer.js:15
    initialize (index):113
    t view.js:16
    t dom_view.js:8
    t renderer.js:11
    WaterfallRendererView (index):110
    build_views build_views.js:20
    build_levels plot_canvas.js:527
    initialize plot_canvas.js:141
    t view.js:16
    t dom_view.js:8
    t plot_canvas.js:29
    build_views build_views.js:20
    build_child_views layout_dom.js:203
    initialize layout_dom.js:24
    t view.js:16
    t dom_view.js:8
    t layout_dom.js:13
    e plot.js:23
    build_views build_views.js:20
    build_child_views layout_dom.js:203
    initialize layout_dom.js:24
    t view.js:16
    t dom_view.js:8
    t layout_dom.js:13
    t box.js:10
    e column.js:7
    i standalone.js:7
    add_document_standalone standalone.js:46
    add_document_from_session server.js:47
    o es6-promise.js:360
    F es6-promise.js:358
    N es6-promise.js:350
    p es6-promise.js:110

index.js:66
```

#### Screenshots or screencasts of the bug in action
![Screenshot from 2019-03-22 15-00-30](https://user-images.githubusercontent.com/8008618/54803566-56c32f80-4cb3-11e9-922b-64b1c56ed1d0.png)

#### Extra (not bad enough for an issue?)

- A README.md for this example would make it easier to execute. For example: "Please install node.js v6.10.0 or higher" and run with command `bokeh serve --show spectrogram/` in parent folder (app/).
- Throw an error that mentions mic data couldn't be grabbed through Pyaudio instead of:

```
  File "/home/*user*/github/bokeh/examples/app/spectrogram/main.py", line 75, in update
    signal, spectrum, bins = audio.data['values']
TypeError: 'NoneType' object is not iterable
```

because of 

```
File "/home/*user*/anaconda3/envs/speechmod/lib/python3.6/site-packages/pyaudio.py", line 441, in __init__
    self._stream = pa.open(**arguments)
OSError: [Errno -9996] Invalid input device (no default output device)
```

**Comment by bryevdv** [comment]
@NumesSanguis after some research it seems the message about `gio` has nothing to do with Bokeh. It is due to some or other deprecated Gnome package being installed on Ubuntu systems, and can be triggered when Python itself opens a webbrowser:

https://bugs.python.org/issue30219

This fix for this is going to be some change in configuration or package updates on your end, there's nothing we can do on our end. Alternatively, don't use `--show` and instead navigate to the app URL manually. 

The JS error is probably not related, but I am marking as `disicussion` until we can actually reproduce things. 

As for the other extras, we would be happy to accept a PR from you that makes those small improvrments. Please do make a separate issue, however. Having multiple concerns or threads of discusion in a one issue makes it much harder to properly track and prioritize things.

**Comment by NumesSanguis** [comment]
@bryevdv Sorry I couldn't experiment with it during the weekend. The `gio` might be indirectly related to bokeh, maybe because of `Tornado` (v5.1.1)? Because executing:

    import webbrowser
    webbrowser.open("https://www.google.com")

in just a terminal (with the same Python environment active) gives no deprecated error.

It seems that the `gio` part is unrelated for the not showing of the Spectrogram though, because removing the `--show` parameter and navigating to the example on both Firefox (Quantum 66.0 (64-bit)) and Chromium (Version 73.0.3683.75 (Official Build) snap (64-bit)) don't give a spectrogram output.
    Note that I'm using the example with synthesized data and not pyaudio for real-time input.

**Comment by bryevdv** [comment]
@NumesSanguis please try things with a recent dev build when you get a chance:

https://bokeh.pydata.org/en/latest/docs/installation.html#developer-builds

1.1 is a big update so it's entirely possible that things were fixed for your situation coincidentally, in which case we can close and skip some very time-intensive effort to reproduce this.

**Comment by NumesSanguis** [comment]
Thanks for your quick reaction, it is working now! I installed bokeh with `conda install -c bokeh/channel/dev bokeh` in a new Anaconda python=3.6 environment (and with `conda install scipy` & `conda install nodejs`). Both Firefox and Chromium work. The `gio deprecation` error is still there when running with `--show` though (but it seems it can be safely ignored).

Just now I also tested it on a Windows 10 64-bit machine and the default conda bokeh is broken, but because of a (`ImportError: cannot import name 'GridBox'`)

This is solved as well in the dev version of Bokeh. So no problem anymore!
